### DISCUSSION TITLE
Rubymine plugin v1.1.17.1262 is freezing Rubymine quite frequently

### DISCUSSION THREAD

**Discussion by ivopt** [community/community#15402] [Copilot Conversations]
I've only updated to the this version of the plugin today (Apr 22nd, 1 day after it has been released) and my rubymine freezes quite often since.

Uninstalling this version and bringing 1.1.16.1219 back makes rubymine stable again.

**Comment by bmuskalla** [comment]
Thanks for the feedback @LynxEyes 

Would you be so kind to see if the same problem happens with the `Nightly` channel as well? You can change the update channel in the Copilot settings and update to a version from there.
And can you share some system details? OS, Arch, Rubymine version, any firewalls/proxy in place?

Thanks for your time

**Reply by ivopt** [reply]
I will try it this coming weekend.

Details: 

* macOS Monterey 12.3.1 (21E258) - intel
* Rubymine 2022.1 (RM-221.5080.196) 
* firewalls/proxy - kind'a, Adguard.

**Reply by bmuskalla** [reply]
Thanks @LynxEyes for the information. We've just released a new version that may fix some of the stated problems. The release should be available within the next 24h. Let us know if your specific problem persists.

**Reply by ivopt** [reply]
Sorry it took me a while to do something about this.

Unfortunately the problem still persists with the newest version of the plugin, I'm rolling back to 2 versions prior. :(

**Reply by bmuskalla** [reply]
Sorry to hear this. Given we have no steps to reproduce, this might be a hard one. Would you be able to state the most recent version you tried and once it freezes, try to get a thread dump from the IDE process? If possible, through `Help > Show Logs...`. And can you describe how you experience the freezes? Is it for a few seconds and resolves itself? If you have any logs that you are willing to share, that'd be great.

**Reply by ivopt** [reply]
Hi, @bmuskalla 

Yes, I recon this would be a hard one to solve. I also think, to make things worse, it might be down to my exact mix of plugins or configs.

To your questions:

When it freezes, it freezes and stays frozen until I kill rubymine via Activity Monitor.
When it freezes, it always seem to go this way: 

1. I either save the file (cmd+s) or I click the run button (to run a rake task, tests or whatever)
2. GH copilot seems to write a suggestion (in that grayed out font) right before the editor inspections run
3. When editor inspections run, then it freezes and stays frozen until I kill it (and I've left it there a few times for quite a while to see if it comes back to life).

Here's my Editor Inspections config:

<img width="1206" alt="image" src="https://user-images.githubusercontent.com/773898/169264470-5f0cd960-855d-4f28-80b6-566e566daa37.png">

As for the logs you're asking, I have provided some to JetBrains on a support issue I opened up with them a bit before opening up this issue - because back then I hadn't made the connection between the GH Copilot plugin and the freezes. JetBrains folks were not able to find anything definitive that pointed to what was wrong.
Only later have I made the connection between the freezes and the GH copilot plugin since if I stick with version 1.1.16.1219 all goes well, if I upgrade to any of the subsequent versions I experience the freezes.

As for the logs, here they are:

[rubymine-logs-20220519-10474612302408139122630391.zip](https://github.com/github/feedback/files/8727992/rubymine-logs-20220519-10474612302408139122630391.zip)

Last freeze I experienced was earlier this week, on the 16th, which are also the logs that have more details on them.
Hope it helps somehow, those look totally alien to me as I'm not used to inspect rubymine logs.
If you folks can find anything that would be awesome.
If not and if I'm the only person experiencing this.. well.. I guess the only sensible thing to do is attribute the fault to my own configuration maybe..

Let me know if I can provide any further data or help in any other way - I may not know how the plugin is made but I'm a developer anyway, I can try to help.. maybe.. I guess..

**Comment by ivopt** [comment]
Well... this stopped happening since a little while back. Since the prior-to-last version of the plugin, I've had no freezes.. so.. done?

**Reply by bmuskalla** [reply]
Great that the fixes did indeed solve this. Thanks for confirming!
